python-semantic-release/python-semantic-release
# feat(default-changelog): add a separate formatted breaking changes section

## Purpose


- Adds clarity for what changes the user must be aware of when bumping to the next major version
- Resolves: #244

## Rationale


This is more of a stepping stone to the next phase where we will drop the need to display the entire commit in the changelog. The Breaking change descriptions are one of the items we didn't display normally even though it was parsed.

## How did you test?


I updated the git repo changelog generator for the E2E scenarios which means the E2E tests that validate changelogs have the correct breaking change summaries included.

I also updated the release notes and changelog template test cases to separate out the breaking change descriptions into their own section and validate the template provides the accurate result.

## How to Verify


1. Checkout this pr
2. Reset by 4 commits to remove the fix
3. Run pytest to see the failures where the commits do not include the breaking change descriptions section as they should
4. Fast forward merge to the latest commit of this pr
5. Run pytest again to see it is fixed.
